src/main/kotlin/app/revanced/patches/music/audio/codecs/patch/CodecsUnlockPatch.kt

```diff
@@ -29,7 +30,14 @@ class CodecsUnlockPatch : BytecodePatch(
 
         val implementation = result.mutableMethod.implementation!!
 
-        val instructionIndex = result.patternScanResult!!.startIndex
+        val instructionIndex = result.patternScanResult!!.startIndex +
+                if (implementation.instructions[result.patternScanResult!!.startIndex - 1].opcode == Opcode.CHECK_CAST) {
```

Comment:
```suggestion
        val scanResultStartIndex = result.patternScanResult!!.startIndex
        val instructionIndex = scanResultStartIndex  +
                if (implementation.instructions[scanResultStartIndex - 1].opcode == Opcode.CHECK_CAST) {
```

Reply:
Also below at line 44 this can be changed too

Reply:
makes no sense for line 44 since this is on another scan result and only used once
however refactor is coming to make this more clear